test/testdata/rewriter/chained_sig.rb

```diff
@@ -0,0 +1,208 @@
+  def bar(a); end
+
+  T::Sig::WithoutRuntime.sig.override { void }.final { void }
+                                             # ^^^^^ error: Cannot add more signature statements after the declaration block
+  def foo; end
+end
+
+class MissingBlocks
+  extend T::Sig
+
+  sig
+# ^^^ error: Signature declarations expect a block
+  def foo; end
+
+  sig.final
+# ^^^^^^^^^ error: Signature declarations expect a block
+    # ^^^^^ error: Method `final` does not exist on `NilClass`
+  def bar; end
+
+  sig.abstract
+# ^^^^^^^^^^^^ error: Signature declarations expect a block
+    # ^^^^^^^^ error: Method `abstract` does not exist on `NilClass`
+  def baz; end
+
+  sig.override
+# ^^^^^^^^^^^^ error: Signature declarations expect a block
+    # ^^^^^^^^ error: Method `override` does not exist on `NilClass`
+  def qux; end
+
+  sig.overridable
+# ^^^^^^^^^^^^^^^ error: Signature declarations expect a block
+    # ^^^^^^^^^^^ error: Method `overridable` does not exist on `NilClass`
+  def quux; end
+end
+
+class InstructionSequenceInSig
+  extend T::Sig
+
+  sig.final { Kernel.puts ""; void }
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Malformed signature: cannot have multiple statements inside a signature block
+    # ^^^^^ error: Method `final` does not exist on `NilClass`
+                            # ^^^^ error: Method `void` does not exist on `T.class_of(InstructionSequenceInSig)`
+  def foo; end
+end
+
+class DuplicateInvocations
+  extend T::Sig
+  extend T::Helpers
+  include Interface
+
+  abstract!
+
+  sig.overridable { overridable.void }
+                  # ^^^^^^^^^^^ error: Duplicate invocation of `overridable` in signature declaration
+  def bar; end
+
+  sig.override { override.void }
+               # ^^^^^^^^ error: Duplicate invocation of `override` in signature declaration
+  def foo; end
+
+  sig.abstract { abstract.void }
+               # ^^^^^^^^ error: Duplicate invocation of `abstract` in signature declaration
+  def baz; end
+end
+
+class DuplicateButMissingBlock
+  extend T::Sig
+  include Interface
+
+  sig.final { void }.override
+                   # ^^^^^^^^ error: Method `override` does not exist on `NilClass`
+  def foo; end
+# ^^^^^^^ error: Method `DuplicateButMissingBlock#foo` implements an abstract method `Interface#foo` but is not declared with `override.`
+end
+
+class InvalidInvocationAfterBlock
+  extend T::Sig
+
+  sig.final { void }.checked(:never)
+                   # ^^^^^^^ error: Cannot use `checked` outside of a sig block
+                   # ^^^^^^^ error: Method `checked` does not exist on `NilClass`
+  def foo; end
+end
+
+module ValidInterface
+  extend T::Sig
+  extend T::Helpers
+
+  interface!
+
+  sig.abstract { void }
+  def foo; end
+
+  sig.abstract { void }
+  def bar; end
+
+  sig.abstract { void }
+  def baz; end
+end
+
+class ValidDoubleChain
+  extend T::Sig
+  include ValidInterface
+
+  sig.override.final { void }
```

Comment:
Can we have a `sig.final.override` test too, please?

We should also ensure that attempting to inherit from `ValidDoubleChain` and override these `final` methods produces errors.